course/categorize.php

```diff
@@ -96,16 +96,42 @@ function flattenarr($ar) {
 		$questionlibs[$row[0]][] = $row[1];
 		$libnames[$row[1]] = $row[2];
 	}
-	$query = "SELECT iq.id,iq.category,iqs.description FROM imas_questions AS iq,imas_questionset as iqs";
+
+	//add assessment names as options
+	//find the names of assessments these questionsetids appear in
+	$query = "SELECT imas_questions.id AS qid,imas_questions.questionsetid AS qsetid,imas_assessments.id AS aid,imas_assessments.name ";
+	$query .= "FROM imas_questions, imas_assessments ";
+	$query .= "WHERE imas_questions.assessmentid=imas_assessments.id ";

```

Comment:
It's equivalent, but I think it's considered slightly more correct to use an explicit JOIN:

````
SELECT (blah) FROM imas_questions INNER JOIN imas_assessments 
ON imas_questions.assessmentid=imas_assessments.id 
WHERE imas_questions.questionsetid = ANY (SELECT ...
````